### ISSUE TITLE
The loss is not get down and the AP is 0.

### ISSUE THREAD

**Issue by SunJJ1996** [facebookresearch/detr#387]
I try to train DETR in my own dataset. The number of classer is 5. I changed the 
`num_classes = 20 if args.dataset_file != 'coco' else 91`
the loss is always in 26~27 since 100 epochs and AP is shown below:
```
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.001
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.002
```
which I have trained about 200 epochs.
How can I fix this problem?
Thank you for your help.

**Comment by qingshanqihai111** [comment]
> I try to train DETR in my own dataset. The number of classer is 5. I changed the
> `num_classes = 20 if args.dataset_file != 'coco' else 91`
> the loss is always in 26~27 since 100 epochs and AP is shown below:
> 
> ```
> IoU metric: bbox
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.001
>  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
>  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.001
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.001
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
>  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.002
> ```
> 
> which I have trained about 200 epochs.
> How can I fix this problem?
> Thank you for your help.

Hello.I have the same problem. Can we talk about it?Would you mind give me your email address?

**Comment by SunJJ1996** [comment]
> > I try to train DETR in my own dataset. The number of classer is 5. I changed the
> > `num_classes = 20 if args.dataset_file != 'coco' else 91`
> > the loss is always in 26~27 since 100 epochs and AP is shown below:
> > ```
> > IoU metric: bbox
> >  Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
> >  Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.001
> >  Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
> >  Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
> >  Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
> >  Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
> >  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.001
> >  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.001
> >  Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.001
> >  Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = -1.000
> >  Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.000
> >  Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.002
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > which I have trained about 200 epochs.
> > How can I fix this problem?
> > Thank you for your help.
> 
> Hello.I have the same problem. Can we talk about it?Would you mind give me your email address?

Well, my email is 963291901@qq.com. But the author is not reply me so I have not figure out that problem yet.

**Comment by llldecade** [comment]
Hello, I met the same problem as you. Have you solved it? Could you give me some suggestions?
Thank you sincerely.

**Comment by alcinos** [comment]
Hi all, 

There can be many reason why your losses are not going down, it's a bit hard to tell without more information. The most likely error is in the data preparation, please check that your boxes are in Coco format, ie [top_left_x, top_left_y, w, h]
It also helps to plot both the predictions and the targets on a few images to see if anything stands out.

You can also consider fine-tuning one of our COCO model (especially if your dataset is small). If you have a different number of classes, you'll have to replace the classification head (search in the issues if you have trouble, this has been asked/answered many times already)

Best of luck

**Comment by zhw2020913** [comment]
You should use  pretrainedweight in  Official website。

![image](https://user-images.githubusercontent.com/71215946/127815958-52dc51e1-b561-410c-8111-28a431eabb0e.png)

**Comment by MyKagome** [comment]
> You should use pretrainedweight in Official website。
> 
> ![image](https://user-images.githubusercontent.com/71215946/127815958-52dc51e1-b561-410c-8111-28a431eabb0e.png)

Hello,my brother,may I add your qq or wechat?Or you can add me,and my qq is 1244353884. I have some problems about the detr and would you mind giving me some advice? Many thanks!

**Comment by mei123456789** [comment]
> You should use pretrainedweight in Official website。
> 
> ![image](https://user-images.githubusercontent.com/71215946/127815958-52dc51e1-b561-410c-8111-28a431eabb0e.png)

I also encountered the same problem, could you add your contact information? I really need your help. If possible, my QQ is 2806609591. Thank you

**Comment by ducvuuit** [comment]
Hi @SunJJ1996, Have you solved the above problem yet? Do you mind sharing it with me?
Thank you in advance.

**Comment by KeyingXu** [comment]
Hi SunJJ1996 and others, I encountered the same issue when adding a new class to original COCO dataset and training from scratch on the merged dataset (COCO dataset + dataset of a new class) without using the pretrained weights. Have you all solved the above problem yet? Could anybody come to explain the solution a little bit? Thanks a lot.
